src/lsst/cmservice/client/loaders.py

```diff
@@ -1,28 +1,373 @@
+
+        file_path = config_values["file_path"]
+        full_file_path = os.path.abspath(os.path.expandvars(file_path))
+        with open(full_file_path, encoding="utf-8") as fin:
+            data = yaml.safe_load(fin)
+
+        if script_template is None:
+            return self._parent.script_template.create(
+                name=key,
+                data=data,
+            )
+
+        return self._parent.script_template.update(
+            row_id=script_template.id,
+            name=key,
+            data=data,
+        )
+
+    def _upsert_specification(
+        self,
+        config_values: dict,
+        *,
+        allow_update: bool = False,
+    ) -> dict:
+        """Upsert and return a Specification
+
+        This will create a new Specification, or update an existing one
+
+        Parameters
+        ----------
+        config_values: dict
+            Values for the ScriptTemplate
+
+        allow_update: bool
+            Allow updating existing templates
+
+        Returns
+        -------
+        out_dict: dict
+            The updated and loaded objects
+        """
+        out_dict: dict[str, list[BaseModel]] = dict(
+            Specification=[],
+            SpecBlockAssocication=[],
+            ScriptTemplateAssociation=[],
+        )
+
+        spec_name = config_values["name"]
+        script_templates = config_values.get("script_templates", [])
+        spec_blocks = config_values.get("spec_blocks", [])
+
+        specification = self._parent.specification.get_row_by_name(spec_name)
+        new_spec = False
+        if specification is None:
+            specification = self._parent.specification.create(name=spec_name)
+
+        for script_list_item_ in script_templates:
+            try:
+                script_template_config_ = script_list_item_["ScriptTemplateAssociation"]
+            except KeyError as msg:
+                raise CMYamlParseError(
+                    f"Expected ScriptTemplateAssociation not {list(script_list_item_.keys())}",
+                ) from msg
+
+            script_template_name = script_template_config_["script_template_name"]
+            alias = script_template_config_.get("alias", script_template_name)
+
+            fullname = f"{spec_name}#{alias}"
+            script_template_assoc = self._parent.script_template_association.get_row_by_fullname(
+                fullname=fullname,
+            )
+            if script_template_assoc and not allow_update:
+                print(f"ScriptTemplateAssociation {fullname} already defined, skipping it")
+                continue
+
+            if script_template_assoc is None:
+                script_template_assoc = self._parent.script_template_association.create(
+                    spec_name=spec_name,
+                    **script_template_config_,
+                )
+                out_dict["ScriptTemplateAssociation"].append(script_template_assoc)
+                continue
+
+            # FIXME
```

Comment:
Was this FIXME fixed just above, and if so, can it go?

Reply:
Oh, nm, I see this is for the update case, not create.  Maybe spare a few more words for posterity about what the problem was that needs to be fixed?

Reply:
I was too lazy to redo the association stuff for the update case, mostly b/c we should probably just ditch the whole SpecBlockAssociation and ScriptTemplateAssociation thing.  It is doing the same thing as the spec_aliases, and that fits better into how we do everything else.